src/util/ip.ts

```diff
@@ -0,0 +1,122 @@
+import { multiaddr, Multiaddr } from "@multiformats/multiaddr";
+import { ENR } from "../enr/index.js";
+
+export type IP = { type: 4 | 6; octets: Uint8Array };
+export type IPUDP = IP & { udp: number };
```

Comment:
@wemeetagain I would do
```ts
interface SocketAddress {
  ip: IP
  port: number
}
```
Or if you really need to clarify it's UDP then
```ts
interface SocketAddressUDP {
  ip: IP
  udp: number
}
```
But I don't see the point, this is used only internally

Reply:
:+1: will refactor